pkg/controller/container-runtime-config/container_runtime_config_controller.go

```diff
@@ -567,14 +567,19 @@ func (ctrl *Controller) syncImageConfig(key string) error {
 			isNotFound := errors.IsNotFound(err)
 			rci := createNewRegistriesConfigIgnition(registriesTOML)
 			if !isNotFound && equality.Semantic.DeepEqual(rci, mc.Spec.Config) {
-				applied = false
-				return nil
+				// if the configuration for the registries is equal, we still need to compare
+				// the generated controller version because during an upgrade we need a new one
+				mcCtrlVersion := mc.Annotations[ctrlcommon.GeneratedByControllerVersionAnnotationKey]
+				if mcCtrlVersion == version.Version.String() {
+					applied = false
+					return nil
+				}
 			}
 			if isNotFound {
 				mc = mtmpl.MachineConfigFromIgnConfig(role, managedKey, &ignv2_2types.Config{})
 			}
 			mc.Spec.Config = createNewRegistriesConfigIgnition(registriesTOML)
-			mc.ObjectMeta.Annotations = map[string]string{
+			mc.Annotations = map[string]string{
```

Comment:
What's the difference here?  It looks like we have `mc.ObjectMeta.Annotations` in another place in this code.

Reply:
none, but ObjectMeta is an embedded field and Go let's you access embedded fields' fields' directly on the top level struct...nothing changes functionality wise

Reply:
I, of course, can change it back...it's just OCD kicking lol

Reply:
ok. moved this back for consistency...